src/main/java/com/scalyr/integrations/kafka/ScalyrSinkConnectorConfig.java

```diff
@@ -138,10 +141,23 @@ public static ConfigDef configDef() {
       if (customAppEventMappings.isEmpty()) {
         throw new ConfigException("No custom event mappings are defined");
       }
+      int numMatchAll = 0;
       for (CustomAppEventMapping mapping : customAppEventMappings) {
-        if (mapping.getMatcherFields().isEmpty() || Strings.isNullOrEmpty(mapping.getMatcherValue())) {
+        if (!mapping.isMatchAll() && (mapping.getMatcherFields().isEmpty() || Strings.isNullOrEmpty(mapping.getMatcherValue()))) {
           throw new ConfigException("Custom event application mapping matcher not defined");
         }
+        if (mapping.isMatchAll()) {
+          numMatchAll++;
+        }
+      }
+
+      // Only one matchAll is allowed to avoid ambiguity
+      if (numMatchAll > 1) {
+        throw new ConfigException("More than one match all custom event mapping matcher is defined");
+      }
+      // Custom event mappings are evaluated in order.  Match all should be last so that other custom event mappings are evaluated.
+      if (numMatchAll == 1 && !customAppEventMappings.get(customAppEventMappings.size() - 1).isMatchAll()) {
+        throw new ConfigException("Match all custom event mapping matcher should be defined last so other custom event mappings can match");
```

Comment:
"Custom event mapper with matchAll=true must be listed last.  Otherwise, some mappers would be ignored."